pkg/steps/template.go

```diff
@@ -856,20 +861,32 @@ func getContainersMap(containers string) map[string]struct{} {
 	return c
 }
 
-func outputAnnotatedContainerLogs(podClient coreclientset.PodInterface, pod *coreapi.Pod) error {
+func artifactsAnnotatedContainerLogs(podClient coreclientset.PodInterface, pod *coreapi.Pod, artifactDir string) error {
+	var validationErrors []error
 	statuses := gatherContainerStatuses(pod)
 	containersToOutput := getContainersMap(pod.ObjectMeta.Annotations[showContainerOutputAnnotation])
+
 	for _, status := range statuses {
 		if s := status.State.Terminated; s != nil {
 			if _, exists := containersToOutput[status.Name]; exists {
-				log.Printf("Container %s logs:", status.Name)
-				if err := printContainerLogsToStdout(podClient, status.Name, pod.Name); err != nil {
-					return fmt.Errorf("%v", err)
+				file, err := os.Create(fmt.Sprintf("%s/%s/%s-out.log", artifactDir, pod.Name, status.Name))
```

Comment:
What creates the `pod.Name` subdirectory of artifact dir? Will it always exist? 

Also, this is not consistent with the description in the documentation change - it does not mention any subdirectory.

Reply:
It will continue for the rest containers. The subdirectory is already being created with `--artifact-dir`

Reply:
Can you provide a link? When is that subdirectory created?

Reply:
https://github.com/openshift/ci-operator/blob/master/pkg/steps/artifacts.go#L117-L121

Reply:
Maybe I am missing something, but I do not see any code that would create a subdirectory for each container there - it simply extracts everything from a tarball, which is created by tarring a content of the `/tmp/artifacts`.

Reply:
@petr-muller Its in https://github.com/openshift/ci-operator/blob/master/pkg/steps/template.go#L136
```go
filepath.Join(s.artifactDir, s.template.Name)
```